lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyDiagnosticParticipant.java

```diff
@@ -22,49 +22,54 @@
 import org.eclipse.lsp4j.Range;
 import org.eclipse.lsp4j.jsonrpc.CancelChecker;
 import io.openliberty.tools.langserver.lemminx.services.FeatureService;
+import io.openliberty.tools.langserver.lemminx.services.LibertyProjectsManager;
 import io.openliberty.tools.langserver.lemminx.services.SettingsService;
 import io.openliberty.tools.langserver.lemminx.util.*;
+
+import java.io.File;
 import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.*;
 
 public class LibertyDiagnosticParticipant implements IDiagnosticsParticipant {
 
     @Override
     public void doDiagnostics(DOMDocument domDocument, List<Diagnostic> diagnostics,
             XMLValidationSettings validationSettings, CancelChecker cancelChecker) {
-        if (!LibertyUtils.isServerXMLFile(domDocument))
+        if (!LibertyUtils.isConfigXMLFile(domDocument))
             return;
         try {
-            validateFeatures(domDocument, diagnostics);
+            validateDom(domDocument, diagnostics);
         } catch (IOException e) {
-            System.err.println("Error validating features");
+            System.err.println("Error validating document " + domDocument.getDocumentURI());
             System.err.println(e.getMessage());
         }
     }
 
-    private void validateFeatures(DOMDocument domDocument, List<Diagnostic> list) throws IOException {
+    private void validateDom(DOMDocument domDocument, List<Diagnostic> list) throws IOException {
         List<DOMNode> nodes = domDocument.getDocumentElement().getChildren();
-        DOMNode featureManager = null;
-        // find <featureManager> element if it exists
+
         for (DOMNode node : nodes) {
             if (LibertyConstants.FEATURE_MANAGER_ELEMENT.equals(node.getNodeName())) {
-                featureManager = node;
-                break;
+                validateFeature(domDocument, list, node);
+            } else if (LibertyConstants.INCLUDE_ELEMENT.equals(node.getNodeName())) {
+                monitorConfigFiles(domDocument, list, node);
             }
         }
-        // No need for validation if there is no <featureManager>
-        if (featureManager == null) {
-            return;
-        }
+        
+    }
 
+    private void validateFeature(DOMDocument domDocument, List<Diagnostic> list, DOMNode node) {
         String libertyVersion =  LibertyUtils.getVersion(domDocument);
 
         final int requestDelay = SettingsService.getInstance().getRequestDelay();
 
         // Search for duplicate features
         // or features that do not exist
         Set<String> includedFeatures = new HashSet<>();
-        List<DOMNode> features = featureManager.getChildren();
+        List<DOMNode> features = node.getChildren();
```

Comment:
I think it would be clearer to still call the parameter featureManager since that is what is passed in.